multiqc/modules/multivcfanalyzer/multivcfanalyzer.py

```diff
@@ -0,0 +1,170 @@
+from collections import OrderedDict
+import logging
+import json
+
+from multiqc.plots import bargraph
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+    """ MultiVCFAnalyzer module """
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='MultiVCFAnalyzer', anchor='multivcfanalyzer',
+        href="https://github.com/alexherbig/MultiVCFAnalyzer",
+        info="""MultiVCFAnalyzer reads multiple VCF files as produced by the GATK UnifiedGenotyper and after filtering provides the combined genotype calls in a number of formats that are suitable for follow-up analyses such as phylogenetic reconstruction, SNP effect analyses, population genetic analyses etc.""")
+
+        # Find and load any MultiVCFAnalyzer reports
+        self.mvcf_data = dict()
+
+        # Find and load JSON file
+        for f in self.find_log_files('multivcfanalyzer', filehandles=True):
+            self.parse_data(f)
+
+        # Filter samples
+        self.mvcf_data = self.ignore_samples(self.mvcf_data)
+
+        # Return if no samples found
+        if len(self.mvcf_data) == 0:
+            raise UserWarning
+
+        # Add in extra columns to data file
+        self.compute_perc_hets()
+
+        # Save data output file
+        self.write_data_file(self.mvcf_data, 'multiqc_multivcfanalyzer_metrics')
+
+        # Add to General Statistics
+        self.addSummaryMetrics()
+
+
+    def parse_data(self, f):
+        try:
+            data = json.load(f['f'])
+        except Exception as e:
+            log.debug(e)
+            log.warn("Could not parse MultiVCFAnalyzer JSON: '{}'".format(f['fn']))
+            return
+
+        # Parse JSON data to a dict
+        for s_name in data:
+            if (s_name == 'metadata'):
+                continue
+            
+            if (s_name == 'metrics'):
+                for sample in data['metrics'].items():
+                    s_clean = sample[0]
+                    if s_clean in self.mvcf_data:
+                        log.debug("Duplicate sample name found! Overwriting: {}".format(s_clean))
+
+                    self.add_data_source(f, s_clean)
+                    self.mvcf_data[s_clean] = dict()
+
+                    for k, v in sample[1].items():
+                        try:
+                            self.mvcf_data[s_clean][k] = float(v)
+                        except ValueError:
+                            self.mvcf_data[s_clean][k] = v
+
+    #Compute % heterozygous snp alleles and add to data
+    def compute_perc_hets(self):
+        """Take the parsed stats from MultiVCFAnalyzer and add one column per sample """
+        for sample in self.mvcf_data:
+            self.mvcf_data[sample]['Heterozygous SNP alleles (percent)'] = (self.mvcf_data[sample]['SNP Calls (het)'] / self.mvcf_data[sample]['SNP Calls (all)'])*100
```

Comment:
Need to catch potential `ZeroDivisionError` exceptions for when there are zero SNPs:

```suggestion
            try:
                self.mvcf_data[sample]['Heterozygous SNP alleles (percent)'] = (self.mvcf_data[sample]['SNP Calls (het)'] / self.mvcf_data[sample]['SNP Calls (all)'])*100
            except ZeroDivisionError:
                self.mvcf_data[sample]['Heterozygous SNP alleles (percent)'] = 'NA'
```

Reply:
And to make this slightly shorter clearer, I would probably write it as follows:

```python
for s_name, metrics in self.mvcf_data.items():
    self.mvcf_data[s_name]['Heterozygous SNP alleles (percent)'] = (metrics['SNP Calls (het)'] / metrics['SNP Calls (all)']) * 100.0
```

Reply:
Note that I made `100` `100.0` in the above example. This is because if your values happen to be `int`s instead of `float`s and you multiple against `100` (another `int`), then your percentages will be round integer numbers.

`100.0` is a `float`, so forces the calculation to work on floats and give you a float result.